### PR TITLE
refactor(sortable-list): improve sortable list accessibility

### PR DESCRIPTION
**Pull request by kherP**
not much info related to accessibility of drag and drop so i refer to this: https://dev.opera.com/articles/accessible-drag-and-drop/

### CODE REVIEW

File: lib/src/SortableList/SortableList.test.tsx [hunk 1]
```diff
@@ -113,4 +120,87 @@ describe("Component: SortableList", () => {
...
+                        Simulate.keyDown(container.querySelector(".drop-container"), { key: "Enter", target: container.querySelectorAll(".sortable-item-wrapper")[2] });
+                    });
+                    act(() => {
+                        Simulate.keyDown(container.querySelector(".drop-container"), { key: "ArrowRight", target: container.querySelectorAll(".sortable-item-wrapper")[2] });
+                    });
+                },
+            },
+            // { statement: "on arrow up button pressed", result: [], triggerEvent: () => Simulate.keyDown(container.querySelector(".stepper-preview"), { key: "ArrowUp" }) },
+            // { statement: "on arrow right button click", result: [], triggerEvent: () => Simulate.keyDown(container.querySelector(".stepper-preview"), { key: "ArrowRight" }) },
+            // { statement: "on home button click", result: [], triggerEvent: () => Simulate.keyDown(container.querySelector(".stepper-preview"), { key: "Home" }) },
```
Comment (eweseong) [inline lib/src/SortableList/SortableList.test.tsx:184]: let's remove commented code

File: lib/src/SortableList/SortableList.tsx [hunk 2]
```diff
@@ -162,6 +163,57 @@ export const SortableList: React.FC<SortableListProps> = ({ onSort, className, d
...
+    );
+
+    const onKeyDown = React.useCallback(
+        (event: React.KeyboardEvent<HTMLDivElement>) => {
+            const selectedItemIndex: number = Number((event.target as HTMLDivElement).dataset?.index);
+            switch (event.key) {
+                case Key.Space:
+                case Key.Enter:
+                    event.preventDefault();
+                    console.log(selectedItemIndex, currentItemIndex);
```
Comment (eweseong) [inline lib/src/SortableList/SortableList.tsx:186]: let's remove debugging code

File: lib/src/SortableList/SortableList.tsx [hunk 3]
```diff
@@ -162,6 +163,57 @@ export const SortableList: React.FC<SortableListProps> = ({ onSort, className, d
...
+                    setCurrentItemIndex((oldItemIndex: number) => (oldItemIndex === null ? selectedItemIndex : null));
+                    break;
+                case Key.ArrowRight:
+                case Key.ArrowDown:
+                    event.preventDefault();
+                    setCurrentItemIndex((oldItemIndex: number) => {
+                        console.log(event.key, oldItemIndex, oldItemIndex !== null, selectedItemIndex, defaultOrders.length - 1);
```
Comment (eweseong) [inline lib/src/SortableList/SortableList.tsx:193]: let's remove debugging code

File: lib/src/SortableList/SortableList.tsx [hunk 4]
```diff
@@ -162,6 +163,57 @@ export const SortableList: React.FC<SortableListProps> = ({ onSort, className, d
...
+        [defaultOrders]
+    );
+
+    const onKeyDown = React.useCallback(
+        (event: React.KeyboardEvent<HTMLDivElement>) => {
+            const selectedItemIndex: number = Number((event.target as HTMLDivElement).dataset?.index);
+            switch (event.key) {
+                case Key.Space:
+                case Key.Enter:
+                    event.preventDefault();
```
Comment (eweseong) [inline lib/src/SortableList/SortableList.tsx:185]: is this needed? this seemed to be preventing the checkbox toggle behaviour within the sortable list

Comment (kherP) [reply lib/src/SortableList/SortableList.tsx]: tested, and you are right xD removed

File: lib/src/SortableList/SortableList.tsx [hunk 5]
```diff
@@ -162,6 +163,54 @@ export const SortableList: React.FC<SortableListProps> = ({ onSort, className, d
         [dragContainerRef, draggingOrders, dragNode, setIsDragging, setCurrentItemIndex, setCurrentItemNode, onSort, setDefaultOrders]
     );
 
+    const onItemSwap = React.useCallback(
+        (currentIndex: number, swapIndex: number, focusTarget: HTMLDivElement) => {
+            const newList: OrderItem[] = defaultOrders.slice(0);
+            const swapItem = newList[swapIndex];
```
Comment (mario-subo) [inline lib/src/SortableList/SortableList.tsx:169]: missing type

File: lib/src/SortableList/sortable-list.scss [hunk 6]
```diff
@@ -52,6 +54,14 @@
     border-radius: 0.25rem;
     border: 1px solid $gray-300;
     padding: 0.25rem 0.75rem;
+    &.on-drag {
+        background-color: $white;
+        position: fixed;
+        top: -$zindex-tooltip;
+        left: -$zindex-tooltip;
```
Comment (mario-subo) [inline lib/src/SortableList/sortable-list.scss:61]: Are you sure the values of top and left should be the z-index values? I don't really get this part.

Comment (kherP) [reply lib/src/SortableList/sortable-list.scss:61]: replace variables with value

**Review by mario-subo — APPROVED** [review]

**Review by eweseong — APPROVED** [review]

**Pull request merged**